dask/array/core.py

```diff
@@ -3213,12 +3213,27 @@ def from_npy_stack(dirname, mmap_mode='r'):
 
 
 def slice_with_dask_array(x, index):
-    y = elemwise(getitem, x, index, dtype=x.dtype)
-
     name = 'getitem-' + tokenize(x, index)
 
-    dsk = {(name, i): k
-           for i, k in enumerate(core.flatten(y._keys()))}
-    chunks = ((np.nan,) * y.npartitions,)
+    if index.ndim < x.ndim:
+        x = x.reshape((-1,) + x.shape[index.ndim:])
+        index = index.flatten()
```

Comment:
This won't work if either array has an unknown length. Not sure of a good, simple way to fix this.

Reply:
This is, I think, something that we want to support.  the use case that I had in mind personally was for machine learning workloads, where the arrays almost always come from dataframes with unknown row-chunk sizes.  

I believe that by default atop will allow nan-chunked arrays to interact with each other assuming that they have equivalent chunk sizes

Reply:
The problem is reshaping doesn't seem to be supported on arrays with unknown dimension lengths. Is that something that could be supported? If not, I don't know of a simple fix.